Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/ListBox.cs

```diff
@@ -0,0 +1,245 @@
+            }
+        }
+    }
+
+    private void SelectLabel(int index)
+    {
+        if (_selectedLabelIndex >= 0)
+        {
+            (Controls[_selectedLabelIndex] as Label)!.TextColor = TextColor;
+            (Controls[_selectedLabelIndex] as Label)!.BackColor = BackgroundColor ?? Color.Transparent;
+        }
+        if (index >= 0)
+        {
+            (Controls[index] as Label)!.TextColor = SelectedTextColor;
+            (Controls[index] as Label)!.BackColor = SelectedRowColor;
+        }
+        _selectedLabelIndex = index;
+    }
+
+    /// <summary>
+    /// Gets or sets the index of the top visible Item
+    /// </summary>
+    public int TopIndex
+    {
+        get => _topIndex;
+        set => SetInvalidatingProperty(ref _topIndex, value);
+    }
+
+    private void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+    {
+        switch (e.Action)
+        {
+            case NotifyCollectionChangedAction.Add:
+                // is the added item visible?
+                if (e.NewStartingIndex < TopIndex + Controls.Count)
+                {
+                    var i = e.NewStartingIndex - TopIndex;
+                    foreach (var item in e.NewItems)
+                    {
+                        (Controls[i] as Label)!.Text =
+                            RowFormatter != null ? RowFormatter(item) : item.ToString();
+                    }
+                }
+                break;
+            case NotifyCollectionChangedAction.Remove:
+                if (e.OldStartingIndex < TopIndex + Controls.Count)
+                {
+                    var i = e.OldStartingIndex - TopIndex;
+
+                    var mustClearSelection = _selectedLabelIndex == i;
+                    var mustMoveSelection = (_selectedLabelIndex > i);
+
+                    while (i < Items.Count && i < Controls.Count - 1)
+                    {
+                        (Controls[i] as Label)!.Text = (Controls[i + 1] as Label)!.Text;
+                        i++;
+                    }
+                    (Controls[i] as Label)!.Text = string.Empty;
+
+                    if (mustClearSelection)
+                    {
+                        SelectLabel(-1);
+                    }
+                    else if (mustMoveSelection)
+                    {
+                        SelectLabel(_selectedLabelIndex - 1);
+                    }
+                }
+                break;
+            case NotifyCollectionChangedAction.Reset:
+                foreach (Label label in Controls)
+                {
+                    label.Text = string.Empty;
+                }
+                break;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override void ApplyTheme(DisplayTheme theme)
+    {
+        foreach (Label label in Controls)
+        {
+            label.ApplyTheme(theme);
+        }
+    }
+
+    /// <inheritdoc/>
+    protected override void OnDraw(MicroGraphics graphics)
```

Comment:
is this override needed?

Reply:
It does because `OnDraw` is an abstract in `Control` so it has to be implemented.